frontend/renew_proxies.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+"""Automatical renewal of proxies necessary for a glideinWMS frontend
+"""
+
+import ConfigParser
+import os
+import pwd
+import re
+import subprocess
+import sys
+
+CONFIG = '/etc/gwms-frontend/proxies.ini'
+
+DEFAULTS = {'use_voms_admin': 'false',
+            'fqan': '/Role=NULL/Capability=NULL',
+            'frequency': '1',
+            'lifetime': '24'}
+
+class Proxy(object):
+    """Class for holding information related to the proxy
+    """
+    def __init__(self, cert, key, output, lifetime):
+        self.cert = cert
+        self.key = key
+        self.output = output
+        self.lifetime = lifetime
+
+    def write(self):
+        """Move output proxy from temp location to its final destination
+        """
+        os.rename(self.output + '.new', self.output)
+
+    def timeleft(self):
+        """Returns the remaining lifetime of the proxy in seconds
+        """
+        stdout, _, _ = _run_command(['voms-proxy-info', '-file', self.output, '-timeleft'])
+        try:
+            return int(stdout)
+        except ValueError:
+            return 0
+
+class VO(object):
+    """Class for holding information related to VOMS attributes
+    """
+    def __init__(self, vo, command, cert=None, key=None):
+        self.name = vo
+        command = command.lstrip('/')
+        self.fqan = '/%s/%s' % (vo, command)
```

Comment:
This looks wrong.  Wouldn't it end up with `/osg/osg/Role=pilot` when the intent is `/osg/Role=pilot`?

However, as a sanity check, I think you *should* ensure that for `VONAME`, if a FQAN is present, it should start with the prefix `/VONAME`.

I.e., an attempt to create a proxy named `/foo/Role=pilot` for the VO `osg` should end in failure.